src/arrayMethodJoin.js

```diff
@@ -4,8 +4,24 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] === undefined && i === this.length - 1) {
+        result += '';
+      } else if (this[i] === null || this[i] === undefined) {
+        result += '';
```

Comment:
this action do nothing, why do you need it?

Reply:
could you do a little brainstorming to avoid redundant conditions? Actually you can solve this task with only two "if" conditions. The first time you need to make sure your current element is not null and undefined, and if not, add it to the result. And then you just have to check if your element is not the last one and add a delimiter if it isn't.